### PR TITLE
Feature/sorting [based on Evgeniy sorting branch]

### PR DESCRIPTION
**Pull request by klumba12**

### CODE REVIEW

File: src/ng/components/head/head.core.js [hunk 1]
```diff
@@ -13,20 +15,68 @@ class HeadCore extends Directive(HEAD_CORE_NAME, {view: `^^${VIEW_CORE_NAME}`})
...
 				}
 			}
 		});
+
+		this.sortToggle = new Command({
+			execute: key => {
+				const sort = this.view.model.sort;
+				const by = Array.from(sort().by);
+				const index = sort.index(by, key);
+				if (index >= 0) {
```
Comment (peaceful-djony) [inline src/ng/components/head/head.core.js:33]: What if logic of change direction move to sort.service.js?

File: src/themes/material/sort.scss [hunk 2]
```diff
@@ -0,0 +1,31 @@
+@import "variables";
+
+table {
+	th:hover {
+		.sort {
+			visibility: visible !important;
```
Comment (titovmx) [inline src/themes/material/sort.scss]: why is here ```!important``` need?

File: src/core/sort/sort.service.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import AppError from 'core/infrastructure/error';
```
Comment (titovmx) [inline src/core/sort/sort.service.js:1]: Add tests for that service would be great

File: src/ng/plugins/groupbar/groupbar.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ class Groupbar extends PluginComponent('qgrid.groupbar.tpl.html') {
 	constructor() {
 		super(...arguments);
 
-		this.newGroup = null;
+		this.newSort = null;
```
Comment (peaceful-djony) [inline src/ng/plugins/groupbar/groupbar.js]: Revert name from newSort to newGroup (for src/ng/plugins/groupbar/groupbar.js file)

File: src/ng/plugins/groupbar/groupbar.html [hunk 5]
```diff
@@ -14,7 +14,7 @@
 		<li>
 			<select ng-disabled="!$groupbar.add.canExecute()"
 					  ng-model="$groupbar.newGroup"
-					  ng-change="$groupbar.add.execute($groupbar.newGroup)">
+					  ng-change="$groupbar.add.execute($groupbar.newSort)">
```
Comment (peaceful-djony) [inline src/ng/plugins/groupbar/groupbar.html]: Revert name from newSort to newGroup (for src/ng/plugins/groupbar/groupbar.html file)

File: src/ng/plugins/groupbar/groupbar.js [hunk 6]
```diff
@@ -15,7 +15,7 @@ class Groupbar extends PluginComponent('qgrid.groupbar.tpl.html') {
 						by: state.by.concat(key)
 					});
 
-					this.newGroup = null;
+					this.newSort = null;
```
Comment (peaceful-djony) [inline src/ng/plugins/groupbar/groupbar.js]: Revert name from newSort to newGroup (for src/ng/plugins/groupbar/groupbar.js file)

File: src/ng/components/head/head.core.js [hunk 7]
```diff
@@ -13,20 +15,68 @@ class HeadCore extends Directive(HEAD_CORE_NAME, {view: `^^${VIEW_CORE_NAME}`})
...
 		});
+
+		this.sortToggle = new Command({
+			execute: key => {
+				const sort = this.view.model.sort;
+				const by = Array.from(sort().by);
+				const index = sortService.index(by, key);
+				if (index >= 0) {
+					const dir = sortService.direction(by[index]);
+					switch (dir) {
```
Comment (titovmx) [inline src/ng/components/head/head.core.js:35]: Can be problem that keys are case sensitive?

File: src/ng/plugins/pivotbar/pivotbar.html [hunk 8]
```diff
@@ -14,7 +14,7 @@
 		<li>
 			<select ng-disabled="!$pivotbar.add.canExecute()"
 					  ng-model="$pivotbar.newGroup"
-					  ng-change="$pivotbar.add.execute($pivotbar.newGroup)">
+					  ng-change="$pivotbar.add.execute($pivotbar.newSort)">
```
Comment (peaceful-djony) [inline src/ng/plugins/pivotbar/pivotbar.html]: Rename newSort to newPivot

File: src/ng/plugins/pivotbar/pivotbar.js [hunk 9]
```diff
@@ -6,7 +6,7 @@ class Pivotbar extends PluginComponent('qgrid.pivotbar.tpl.html') {
 	constructor() {
 		super(...arguments);
 
-		this.newGroup = null;
+		this.newSort = null;
```
Comment (peaceful-djony) [inline src/ng/plugins/pivotbar/pivotbar.js]: Rename newSort to newPivot

File: src/ng/components/head/head.core.js (hunk 7 above)
Comment (klumba12) [reply src/ng/components/head/head.core.js:35]: Don't think so exception will be thrown and this is ok, lets give to user one way to do smth :)

**Review by titovmx — APPROVED** [review]

**Pull request merged**